datashield/src/main/java/org/molgenis/datashield/pojo/DataShieldCommand.java

```diff
@@ -0,0 +1,89 @@
+package org.molgenis.datashield.pojo;
+
+import static java.time.Instant.now;
+import static org.molgenis.datashield.pojo.DataShieldCommand.DataShieldCommandStatus.*;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public class DataShieldCommand<T> {
+  private final Instant createDate;
+  private final UUID id;
+  private final String expression;
+  private final boolean withResult;
+  @SuppressWarnings("java:S3077") // CompletableFuture is thread-safe
```

Comment:
Nice, these are the properties that need volatile I pressume.

Reply:
Yes, I think so, the ones that are mutable should be synchronized across threads